src/communication_interface.cc

```diff
@@ -302,13 +304,15 @@ void CommunicationInterface::HandlePlan(
   dexai::log()->info("CommunicationInterface::HandlePlan: Received new plan {}",
                      robot_spline->utime);
 
+  const auto current_mode {GetRobotMode()};
+
   //$ check if in proper mode to receive commands
-  if (!CanReceiveCommands()) {
-    dexai::log()->error(
-        "CommunicationInterface::HandlePlan: "
-        "Discarding plan with utime: {},"
-        " robot is in wrong mode!",
-        robot_spline->utime);
+  if (!CanReceiveCommands(current_mode)) {
+    const auto err_msg {fmt::format(
+        "Discarding plan with utime: {}, robot is in wrong mode: {}!",
+        robot_spline->utime, utils::RobotModeToString(current_mode))};
+    dexai::log()->error("CommunicationInterface::HandlePlan: {}", err_msg);
+    PublishDriverStatus(false, err_msg);
     return;
```

Comment:
Could we instead try and continuously publish mode to a channel? We already publish a bool indicating whether the robot is user stopped and detect this in dig. If we expand that we can reuse/expand on the same interface

Reply:
Add a `<robot>_BRAKES_LOCKED` bool message